### PR TITLE
Reject invalid pattern lengths

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

File: components/datetime/src/format/neo.rs [hunk 1]
```diff
@@ -2031,139 +2034,175 @@ impl<R: DateTimeNamesMarker> RawDateTimeNames<R> {
...
+                // m..mm
+                (FS::Minute, One | TwoDigit) => load_fdf = true,
+
+                // s..ss
+                (FS::Second(Second::Second), One | TwoDigit) => load_fdf = true,
+
+                // A
+                (FS::Second(Second::Millisecond), _) => load_fdf = true,
+
+                // ???
```
Comment (Manishearth) [inline components/datetime/src/format/neo.rs:2195]: nit: ?????

(more docs please if this isn't following the regular scheme)

**Review by Manishearth — APPROVED** [review]

File: components/datetime/src/format/neo.rs (hunk 1 above)
Comment (robertbastian) [reply components/datetime/src/format/neo.rs:2195]: yeah @sffc, I don't understand where this is coming from. There's no symbol associated with that type, and "decimal second" is not a UTS-35 concept

Comment (Manishearth) [reply components/datetime/src/format/neo.rs:2195]: I'm happy with this landing for now with someone doing the comment fix in a followup, but probably worth waiting for @sffc review too

Comment (sffc) [reply components/datetime/src/format/neo.rs:2195]: DecimalSecond was added in https://github.com/unicode-org/icu4x/pull/5392 and it is a single field to represent things like `ss.SSS` in UTS 35 patterns. Previously it was handled with hacky, buggy code.

**Pull request merged**